webgui/InstallGUI.cmake

```diff
@@ -2,7 +2,31 @@ message("Install npm packages...")
 
 execute_process(
   COMMAND cp ${CC_PACKAGE} ${INSTALL_SCRIPTS_DIR}/package.json
+  WORKING_DIRECTORY ${INSTALL_SCRIPTS_DIR})
+
+execute_process(
   COMMAND npm install
   WORKING_DIRECTORY ${INSTALL_SCRIPTS_DIR})
 
-message("Installation of npm packages are finished.")
\ No newline at end of file
+execute_process(
+  COMMAND ${CMAKE_COMMAND} -E create_symlink dojo-util node_modules/util
+  WORKING_DIRECTORY ${INSTALL_SCRIPTS_DIR})
+
+message("Installation of npm packages are finished.")
+
+message("Initiating the Dojo Build System...")
+
+if (CMAKE_BUILD_TYPE STREQUAL "Release")
+  set(DOJO_OPTIMIZE "")
+  set(DOJO_LAYEROPTIMIZE "closure")
+else()
+  set(DOJO_OPTIMIZE "")
+  set(DOJO_LAYEROPTIMIZE "")
+endif()
+configure_file(${CC_PROFILE} ${INSTALL_SCRIPTS_DIR}/app.profile.js)
+
+execute_process(
+  COMMAND node_modules/util/buildscripts/build.sh --profile app.profile.js
```

Comment:
It is installed with the *dojo-util* npm package.

Reply:
For the recoed: the `util` directory is a symlink created during the installation phase of CodeCompass, because the Dojo Build System expects it to be installed there (it is hard-coded). However, *npm* does not support for a package to be in a directory other than its name. Hence the symlink. (Dojo therefore recommends *bower* to be used, where it is supported, but I did not want to change that much.)